src/app/Trade.ts

```diff
@@ -19,9 +19,9 @@ export class Trade {
    * @returns true if transaction will be processed, and false otherwise
    */
   processOffer(
-    offer: Offer,
+    offer: Offer[],
```

Comment:
Perhaps we could pass the animals value here already

Reply:
it would need to be converted into another type. is it really necessary?

Reply:
We should be getting information from where it really is. Values are passed in to create Animal.

Reply:
I've removed creation animal objects from trade. However this is necessary to pass here Offer array with animal names and quantity in process to calculate value and possibly reduce it to optimal value.